src/components/side-panel/layer-panel/vis-config-by-field-selector.js

```diff
@@ -50,17 +51,30 @@ export default class VisConfigByFieldSelector extends Component {
   };
 
   render() {
-    const {property, showScale, selectedField, description, scaleOptions = []} = this.props;
+    const {property, showScale, selectedField, description, intl, scaleOptions = []} = this.props;
 
     return (
       <SidePanelSection>
         <SidePanelSection>
           <PanelLabelWrapper>
             <PanelLabel>
-              {this.props.label || `${capitalizeFirstLetter(property)} based on`}
+              {(this.props.label && (
+                <FormattedMessage id={`property.${camelize(this.props.label)}`} />
```

Comment:
The `camelize` transform now needs to be language specific. Does it only make sense for languages that are Latin based?

Reply:
I added `defaultMessage: label` and `defaultMessage: property`. That way the original value will be used if camelized string is not found from the localization files. Do you think that it will be sufficient?